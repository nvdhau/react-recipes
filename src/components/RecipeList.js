import React from 'react';
import PropTypes from 'prop-types';
import RecipeListItem from './RecipeListItem';

// py: padding bot and top
// [1]
// const RecipeList = ({ style, recipes, onClick, onFavorited, favorites
// }) => (
const RecipeList = ({
  style,
  recipes,
  favorites,
  ...props // = [1]
}) => (
  <ul className="list-reset">
    {recipes.map(recipe => (
      <RecipeListItem
        key={recipe.id}
        favorited={favorites.includes(recipe.id)}
        recipe={recipe}
        // [1]
        // onClick={onClick}
        // onFavorited={onFavorited}
        {...props}
      />
    ))}
  </ul>
);

RecipeList.propTypes = {
  style: PropTypes.object,
  recipes: PropTypes.array,
  onClick: PropTypes.func,
  onFavorited: PropTypes.func,
  favorites: PropTypes.array,
};

RecipeList.defaultProps = {
  recipes: [],
  favorites: [],
};

export default RecipeList;
