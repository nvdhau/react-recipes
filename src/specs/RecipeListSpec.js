import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../components/RecipeList';

const testRecipes = [
  {
    id: 1,
    name: 'Recipe 1',
    category: 'Category 1',
  },
  {
    id: 2,
    name: 'Recipe 2',
    category: 'Category 2',
  },
  {
    id: 3,
    name: 'Recipe 3',
    category: 'Category 3',
  },
];

// think component as IO function
describe('<RecipeList />', () => {
  test('Should not break when no recipes passed', () => {
    const component = renderer.create(<RecipeList />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly', () => {
    const component = renderer.create(<RecipeList recipes={testRecipes} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render favorited state correctly', () => {
    const component = renderer.create(
      <RecipeList recipes={testRecipes} favorites={[1]} />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
