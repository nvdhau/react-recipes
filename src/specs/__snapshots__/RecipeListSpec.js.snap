// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> Should not break when no recipes passed 1`] = `
<ul
  className="list-reset"
/>
`;

exports[`<RecipeList /> Should render correctly 1`] = `
<ul
  className="list-reset"
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    id={1}
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      🤍
    </span>
    <span>
      Recipe 1
    </span>
    <span>
      Category 1
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    id={2}
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      🤍
    </span>
    <span>
      Recipe 2
    </span>
    <span>
      Category 2
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    id={3}
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      🤍
    </span>
    <span>
      Recipe 3
    </span>
    <span>
      Category 3
    </span>
  </li>
</ul>
`;

exports[`<RecipeList /> Should render favorited state correctly 1`] = `
<ul
  className="list-reset"
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    id={1}
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      💗
    </span>
    <span>
      Recipe 1
    </span>
    <span>
      Category 1
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    id={2}
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      🤍
    </span>
    <span>
      Recipe 2
    </span>
    <span>
      Category 2
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    id={3}
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      🤍
    </span>
    <span>
      Recipe 3
    </span>
    <span>
      Category 3
    </span>
  </li>
</ul>
`;
